### PR TITLE
write csv to pg

### PR DESCRIPTION
**Pull request by anacollado**

### CODE REVIEW

File: database/writeFileToDb.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const pg = require('pg');
```
Comment (bkwon94) [inline database/writeFileToDb.js:1]: Oh what is pg used for? Also this is for postgreSQL right?

File: database/writeFileToDb.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//'books' db already created, included in conn string
+const CONNECTION_STRING = process.env.BOOKS_PG_CONN_STRING;
+
+const pool = new pg.Pool({
+  connectionString: CONNECTION_STRING
+})
+
+pool.query(`CREATE TABLE IF NOT EXISTS details (
```
Comment (bkwon94) [inline database/writeFileToDb.js:9]: Maybe prepare your query in a variable and then pass it in here.

File: database/writeFileToDb.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  characters JSON,
+  settings JSON,
+  litAwards JSON,
+  editions JSON
+)`, (err, results) => {
+  if (err) {
+    console.error('error after table creation:', err);
+  }
+  console.log('about to run copy');
+  pool.query(`COPY details(type,pageNum,publisher,dates,title,isbn10,isbn13,language,characters,settings,litAwards,editions) FROM '/Users/anacollado/Desktop/hr-projects/sdc-book-details-service/bookDetails.csv' WITH (FORMAT CSV, DELIMITER '|')`,
```
Comment (bkwon94) [inline database/writeFileToDb.js]: Not sure if possible but Im sure you can utilize __dirname in your path

**Review by bkwon94 — APPROVED** [review]

File: database/seed.js [hunk 4]
```diff
@@ -1,108 +1,58 @@
...
-const { writeRecordsToFile } = require('./seedMethods');
+const { createBook } = require('./sampleDataModel');
+const fs = require('fs');
+const fastCSV = require('fast-csv');
 
-const ONE_MILLION = 1000000;
-
-writeRecordsToFile(ONE_MILLION * 10);
 
+const ONE_MILLION = 1000000;
```
Comment (therobinkim) [inline database/seed.js:7]: i like this! helps make sure there isn't an extra 0 (or too few 0s)

File: database/writeFileToDb.js (hunk 3 above)
Comment (therobinkim) [reply database/writeFileToDb.js]: agreeing w/ @bkwon94; maybe use `path` to generate the file path as this line of code is hyper specific to your machine (and therefore wouldn't work on anybody else's computer)

File: database/seed.js [hunk 5]
```diff
@@ -1,108 +1,58 @@
...
+      nextBook.characters,
+      nextBook.settings,
+      nextBook.litAwards,
+      nextBook.editions,
+    ]
+    let canContinue = csvWriteStream.write(
+      bookRow
+    );
+    if (!canContinue) {
+      csvWriteStream.once('drain', writeNextRecordToCSV);
```
Comment (therobinkim) [inline database/seed.js:46]: would moving ln 46 to before ln 42 help eliminate lns 45-49?

File: database/seed.js [hunk 6]
```diff
@@ -1,108 +1,58 @@
...
-// });
 
-// const db = Promise.promisifyAll(connection, { multiArgs: true });
+const writeRecordsToFile = (size) => {
+  const LABEL = 'CSV data writing';
+  console.time(LABEL);
+  //use '|' delimiter to avoid conflict with fields with commas
+  const csvWriteStream = fastCSV.createWriteStream({delimiter: '|'});
+  const fsWriteStream = fs.createWriteStream("bookDetails.csv");
+  fsWriteStream.on("finish", () => {
```
Comment (therobinkim) [inline database/seed.js:16]: would recommend turning on your linter so you can be consistent w/ small details like `'` vs `"`

**Pull request merged**